### DISCUSSION TITLE
How to associate touch coordinates with components in an MCU?

### DISCUSSION THREAD

**Discussion by Song-aff** [slint-ui/slint#5099] [Q&A] [answered]
https://github.com/slint-ui/slint/blob/62bd0f37406fcc7e263405247582c416ca316441/api/rs/slint/mcu.md?plain=1#L213

**Comment by ogoffart** [comment, marked as answer]
You ask the touch driver.

There are some example in the repository for a few different touch screens:


https://github.com/slint-ui/slint/blob/c89af275e5cca92f42f3710e023ace458e1051b1/examples/mcu-board-support/pico_st7789.rs#L256-L279

https://github.com/slint-ui/slint/blob/c89af275e5cca92f42f3710e023ace458e1051b1/examples/mcu-board-support/stm32h735g.rs#L356-L371

https://github.com/slint-ui/slint/blob/c89af275e5cca92f42f3710e023ace458e1051b1/examples/mcu-board-support/esp32_s3_box.rs#L139-L174

But this is going to be dependent of your touch driver.

I hope this helps.

**Reply by Song-aff** [reply]
Can I understand it this way? Define a touch event, provide the position, and then trigger it through dispatch_event. I don't need to record the position of a component on the screen.
`let button = slint::platform::PointerEventButton::Left; `
`WindowEvent::PointerPressed { position, button }`
`window.dispatch_event(event);`

**Reply by ogoffart** [reply]
Yes, exactly.

(And if there is are no touch or button pressed, you don't need to dispatch an event)
